File: go/fibonacci_sequence/fibonacci_sequence_13.go
// Package main implements a simple program to generate Fibonacci sequence.
// The purpose of this program is to provide a detailed example of how to use global variables
// and to demonstrate the Fibonacci sequence generation in Go.
package main

import "fmt"

// Global variable to store the Fibonacci sequence
var fibSequence []int

// Global variable to store the number of terms in the Fibonacci sequence
var numTerms int

// Function to initialize the global variables
func init() {
    // Initialize the number of terms in the Fibonacci sequence
    numTerms = 10
    // Initialize the Fibonacci sequence slice with the appropriate length
    fibSequence = make([]int, numTerms)
}

// Function to generate the Fibonacci sequence
func generateFibonacci() {
    // The first two terms of the Fibonacci sequence are always 0 and 1
    fibSequence[0] = 0
    fibSequence[1] = 1

    // Loop to generate the remaining terms of the Fibonacci sequence
    for i := 2; i < numTerms; i++ {
        // Each term is the sum of the two preceding terms
        fibSequence[i] = fibSequence[i-1] + fibSequence[i-2]
    }
}

// Main function to execute the program
func main() {
    // Call the function to generate the Fibonacci sequence
    generateFibonacci()

    // Print the generated Fibonacci sequence
    fmt.Println("Fibonacci sequence:")
    for i := 0; i < numTerms; i++ {
        fmt.Printf("%d ", fibSequence[i])
    }
    fmt.Println()
}

File: vbnet/fibonacci_sequence/fibonacci_sequence_13.vb
' This program calculates the Fibonacci sequence up to a specified number of terms.
' The purpose of this program is to help programmers understand how to implement the Fibonacci sequence in Visual Basic .NET.
' The program uses a global variable to store the current term in the sequence.

Module FibonacciSequence

    ' Global variable to store the current term in the Fibonacci sequence
    Dim currentTerm As Integer

    Sub Main()
        ' Declare variables to store the number of terms and the first two terms of the sequence
        Dim numTerms As Integer
        Dim firstTerm As Integer = 0
        Dim secondTerm As Integer = 1

        ' Prompt the user to enter the number of terms
        Console.WriteLine("Enter the number of terms:")
        numTerms = Convert.ToInt32(Console.ReadLine())

        ' Print the first two terms of the sequence
        Console.WriteLine("Fibonacci Sequence:")
        Console.WriteLine(firstTerm)
        Console.WriteLine(secondTerm)

        ' Loop to calculate and print the remaining terms of the sequence
        For i As Integer = 3 To numTerms
            ' Calculate the current term by adding the previous two terms
            currentTerm = firstTerm + secondTerm

            ' Print the current term
            Console.WriteLine(currentTerm)

            ' Update the previous two terms
            firstTerm = secondTerm
            secondTerm = currentTerm
        Next

        ' Wait for user input before closing the program
        Console.ReadLine()
    End Sub

End Module

